byterun/sys.c

```diff
@@ -183,15 +182,23 @@ CAMLprim value caml_sys_open(value path, value vflags, value vperm)
   int fd, flags, perm;
   char * p;
 
+#if defined(O_CLOEXEC)
+  flags = O_CLOEXEC;
+#elif defined(_WIN32)
+  flags = _O_NOINHERIT;
+#else
+  flags = 0;
+#endif
+
   caml_sys_check_path(path);
   p = caml_stat_strdup(String_val(path));
-  flags = caml_convert_flag_list(vflags, sys_open_flags);
+  flags |= caml_convert_flag_list(vflags, sys_open_flags);
   perm = Int_val(vperm);
   /* open on a named FIFO can block (PR#1533) */
   caml_enter_blocking_section();
   fd = CAML_SYS_OPEN(p, flags, perm);
   /* fcntl on a fd can block (PR#5069)*/
-#if defined(F_SETFD) && defined(FD_CLOEXEC)
+#if defined(F_SETFD) && defined(FD_CLOEXEC) && !defined(O_CLOEXEC)
```

Comment:
Shouldn't this also say `&& !defined(_WIN32)` ?

Reply:
It could for belt 'n braces, but Windows is never going to define fcntl or `F_SETFD` (https://msdn.microsoft.com/en-us/library/windows/desktop/ms740126(v=vs.85).aspx)

Reply:
I would say for clarity rather than belt and braces. My preferred solution would be to have only one place where you test the conditions and define both the default value for `flags` and a macro used to disable the `fcntl` call.